File: OpenDataAccess/OpenDataAccess.Service/Request.cs
using System;
using System.Collections.Generic;
using System.Collections.Specialized;
using System.Linq;
using System.Text;
using System.Web;
using OpenDataAccess.Entity;
using OpenDataAccess.Data;
using System.IO;

namespace OpenDataAccess.Service
{
    public class Request
    {
        public Guid RequestId { get; private set; }
        public string Content { get; set; }
        public Dictionary<string, List<IEntityData>> Entities { get; set; }
        public EntityType Entity { get; set; }
        public EntityType OperationLogEntity { get; set; }
        public string MethodName { get; set; }
        public string RequestType { get; set; }
        public NameValueCollection QueryString { get; set; }
        public string EntityName { get; set; }
        public string OperationUser { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public long ElapsedMilliseconds { get; set; }
        public bool IsLog { get; set; }
        public bool IsPostQuery { get; set; }
        public string IPAddress { get; set; }
        public Action<TextWriter, string> CustomWriterLog { get; set; }
        public Exception Excption { get; set; }
        public string RootPath { get; set; }
        public string PathAndQuery { get; set; }
        public string RawUrl { get; set; }
        public string PathInfo { get; set; }

        public Request()
        {
            this.RequestId = Guid.NewGuid();
        }
    }
}

File: JavaOpenDataAccess/src/OpenDataAccess/Service/Request.java
package OpenDataAccess.Service;

import OpenDataAccess.Entity.EntityType;
import OpenDataAccess.Entity.IEntityData;
import OpenDataAccess.LambdaInterface.IAction2;
import OpenDataAccess.LambdaInterface.IFunction;
import OpenDataAccess.Utility.Common;
import com.sun.jndi.toolkit.url.UrlUtil;

import java.io.BufferedWriter;
import java.net.MalformedURLException;
import java.util.Date;
import java.util.List;
import java.util.Map;

public class Request {
    public String GetRequestId() {
        return _RequestId;
    }

    private String _RequestId = null;
    public String Content = null;
    public Map<String, List<IEntityData>> Entities = null;
    public EntityType Entity = null;
    public EntityType OperationLogEntity = null;
    public String MethodName = null;
    public String RequestType = null;
    public Map<String, String> QueryString = null;
    public String EntityName = null;
    public String OperationUser = null;
    public Date StartTime = null;
    public Date EndTime = null;
    public long ElapsedMilliseconds = 0;
    public boolean IsLog = false;
    public boolean IsPostQuery = false;
    public String IPAddress = null;
    public IAction2<BufferedWriter, String> CustomWriterLog = null;
    public Exception Excption = null;
    public String RootPath = null;
    public String PathAndQuery = null;
    public String RawUrl = null;
    public String PathInfo = null;

    public IFunction<Boolean> IsDirectRequest = null;

    public Request() {
        this._RequestId = Common.CreateGuid();
    }

    public String GetParameterValue(String name) throws MalformedURLException {
        String key = QueryString.keySet().stream().filter(f -> f.trim().toLowerCase().equals(name.trim().toLowerCase())).findFirst().get();
        if (Common.StringIsNullOrEmpty(key)) {
            String value = QueryString.get(key);
            if (value.equals("undefined")) return "";
            else return UrlUtil.decode(key);
        } else return "";
    }
}
